drivers/flash/soc_flash_mcux.c

```diff
@@ -376,6 +393,15 @@ static int flash_mcux_init(const struct device *dev)
 
 	rc = FLASH_Init(&priv->config);
 
+#if defined(FTFx_DRIVER_IS_FLASH_RESIDENT) && FTFx_DRIVER_IS_FLASH_RESIDENT
+	/* MCUXSDK FTFX driver's commadAddr is an address to data (LSB = 0), but
+	 * (uint32_t)flash_ftfx_run_command is an address to code (LDB = 1), so
+	 * clear the LSB here.
+	 */
+	priv->config.ftfxConfig->runCmdFuncAddr.commadAddr =
+		((uint32_t)flash_ftfx_run_command) & ~0x01U;
```

Comment:
@zejiang0jason , I am confused by this code. Are you assigning this to be a function pointer?

Reply:
Hi @mmahadevan108 , MCUX SDK flash driver's `commandAddr` is an `uint32_t` type address:
https://github.com/nxp-mcuxpresso/mcuxsdk-core/blob/main/drivers/flash/fsl_ftfx_controller.h#L347.

Here `flash_ftfx_run_command` is a function address whose LSB is 1.
SDK flash driver's `commandAddr` assumes the LSB is 0, https://github.com/nxp-mcuxpresso/mcuxsdk-core/blob/main/drivers/flash/fsl_ftfx_controller.c#L1342. So need to clear LSB.